src/structures/idt.rs

```diff
@@ -444,65 +445,59 @@ impl InterruptDescriptorTable {
     }
 }
 
-impl Index<usize> for InterruptDescriptorTable {
-    type Output = Entry<HandlerFunc>;
+impl<I: RangeBounds<usize>> Index<I> for InterruptDescriptorTable {
+    type Output = [Entry<HandlerFunc>];
 
     /// Returns the IDT entry with the specified index.
     ///
     /// Panics if index is outside the IDT (i.e. greater than 255) or if the entry is an
     /// exception that pushes an error code (use the struct fields for accessing these entries).
-    fn index(&self, index: usize) -> &Self::Output {
-        match index {
-            0 => &self.divide_by_zero,
-            1 => &self.debug,
-            2 => &self.non_maskable_interrupt,
-            3 => &self.breakpoint,
-            4 => &self.overflow,
-            5 => &self.bound_range_exceeded,
-            6 => &self.invalid_opcode,
-            7 => &self.device_not_available,
-            9 => &self.coprocessor_segment_overrun,
-            16 => &self.x87_floating_point,
-            18 => &self.machine_check,
-            19 => &self.simd_floating_point,
-            20 => &self.virtualization,
-            i @ 32..=255 => &self.interrupts[i - 32],
-            i @ 15 | i @ 31 | i @ 21..=29 => panic!("entry {} is reserved", i),
-            i @ 8 | i @ 10..=14 | i @ 17 | i @ 30 => {
-                panic!("entry {} is an exception with error code", i)
-            }
-            i => panic!("no entry with index {}", i),
+    fn index(&self, index: I) -> &Self::Output {
+        let lower_idx = match index.start_bound() {
+            Included(start) => *start,
+            Excluded(start) => *start + 1,
+            Unbounded => 32,
```

Comment:
I would prefer to error in this case because it would be confusing that `..256` is a smaller range than `0..256`.